### PR TITLE
GEODE-7868: Enhance ReflectionBasedAutoSerializer

### PR DESCRIPTION
**Pull request by mmartell**
Enhances the ReflectionBasedAutoSerializer to support unsigned types, Guids, and Decimals. There are no new APIs, only additions to the autoserializer.

New AutorSerializer tests have been added to the new test framework for .NET.

### CODE REVIEW

File: clicache/src/ReflectionBasedAutoSerializer.cpp [hunk 1]
```diff
@@ -228,18 +254,30 @@ namespace Apache
                 break;
 			          case FieldType::BYTE:
                   return r->ReadByte(m_fieldName);
+                break;
+			          case FieldType::SBYTE:
+                  return r->ReadByte(m_fieldName);
```
Comment (pivotal-jbarrett) [inline clicache/src/ReflectionBasedAutoSerializer.cpp:259]: This didn't require any explicit casting like the others?

File: clicache/src/ReflectionBasedAutoSerializer.cpp [hunk 2]
```diff
@@ -288,268 +338,299 @@ namespace Apache
                 break;
 			          case FieldType::ARRAY_OF_BYTE_ARRAYS:
                   return r->ReadArrayOfByteArrays(m_fieldName);
+                break;
+			          case FieldType::GUID:
+                  return gcnew Guid(r->ReadByteArray(m_fieldName));
```
Comment (pivotal-jbarrett) [inline clicache/src/ReflectionBasedAutoSerializer.cpp:343]: Awesome solution!

**Comment by moleske** [comment]
The LGTM failed because it couldn't download `https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar`.  However, Travis successfully downloaded `https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.gz`

I'm not sure why one ends with `tar` (the wrong ending) and the other ends with `tar.gz` (the right ending)

File: clicache/src/ReflectionBasedAutoSerializer.cpp (hunk 1 above)
Comment (mmartell) [reply clicache/src/ReflectionBasedAutoSerializer.cpp:259]: No. Remember ReadByte returns an SByte already. A very bogus API name for a .NET developer. This is one of the main reasons we need to clean up the existing API.

**Pull request merged**